packages/kit/src/vite/plugin.js

```diff
@@ -0,0 +1,368 @@
+import { svelte as svelte_plugin } from '@sveltejs/vite-plugin-svelte';
+import fs from 'fs';
+import colors from 'kleur';
+import path from 'path';
+import { searchForWorkspaceRoot } from 'vite';
+import { mkdirp, posixify, rimraf } from '../utils/filesystem.js';
+import * as sync from '../core/sync/sync.js';
+import { build_server } from '../core/build/build_server.js';
+import { build_service_worker } from '../core/build/build_service_worker.js';
+import { prerender } from '../core/build/prerender/prerender.js';
+import { print_config_conflicts, process_config } from '../core/config/index.js';
+import { configure_server } from '../core/dev/plugin.js';
+import { generate_manifest } from '../core/generate_manifest/index.js';
+import { get_aliases, get_runtime_path, logger, resolve_entry } from '../core/utils.js';
+import { deep_merge } from '../utils/object.js';
+import { find_deps, get_default_config } from '../core/build/utils.js';
+
+const cwd = process.cwd();
+
+/**
+ * @param {import('types').ValidatedConfig} svelte_config
+ * @return {import('vite').Plugin}
+ */
+export const sveltekit = function (svelte_config) {
```

Comment:
this needs to take a `Config['kit']`, not a `ValidatedConfig` (or rather, `Config` becomes what's currently `Config['kit']`, and mutatis mutandis for `ValidatedConfig`). we'll have to do the validation inside the plugin

```suggestion
/**
 * @param {import('types').Config['kit']} svelte_config
 * @return {import('vite').Plugin}
 */
export const sveltekit = function (svelte_config = {}) {
```